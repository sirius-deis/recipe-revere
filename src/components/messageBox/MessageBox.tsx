import { FC, PropsWithChildren } from "react";
import styles from "./MessageBox.module.css";
import Button from "../button/Button";
import Panel from "../panel/Panel";

interface MessageBoxProps {
  closeMessageBox: () => void;
}

const MessageBox: FC<PropsWithChildren & MessageBoxProps> = ({
  children,
  closeMessageBox,
}) => {
  return (
    <div className={styles.container}>
      <Panel direction="horizontal" centered>
        {children}
        <div className={styles["button-container"]}>
          <Button size="sm" onClick={closeMessageBox}>
            &#10008;
          </Button>
        </div>
      </Panel>
    </div>
  );
};

export default MessageBox;
