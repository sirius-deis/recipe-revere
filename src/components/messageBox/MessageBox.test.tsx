import { render, screen, fireEvent } from "@testing-library/react";
import MessageBox from "./MessageBox";

describe("MessageBox component", () => {
  it("should match snapshot", () => {
    const fn = jest.fn();
    const { container } = render(
      <MessageBox closeMessageBox={fn}>
        <div>Some element</div>
      </MessageBox>
    );
    expect(container).toMatchSnapshot();
  });

  it("should close the message box", () => {
    const fn = jest.fn();
    render(
      <MessageBox closeMessageBox={fn}>
        <div>Some element</div>
      </MessageBox>
    );

    fireEvent.click(screen.getByRole("button"));
    expect(fn).toHaveBeenCalled();
    expect(fn).toHaveBeenCalledTimes(1);
  });
});
