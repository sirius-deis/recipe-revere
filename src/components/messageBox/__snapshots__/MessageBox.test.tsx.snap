// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBox component should match snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <div>
      Some element
    </div>
    <div
      class="button-container"
    >
      <button
        class="btn btn-sm btn-main"
      >
        âœ˜
      </button>
    </div>
  </div>
</div>
`;
